lalrpop/src/normalize/prevalidate/mod.rs

```diff
@@ -167,7 +167,21 @@ impl<'grammar> Validator<'grammar> {
                         sym);
                 }
             }
-            Symbols::Anon(_) => { }
+            Symbols::Anon(syms) => { 
+                let empty_string = "".to_string();
+                let action = {
+                    match alternative.action {
+                        Some(ActionKind::User(ref action)) => action,
+                        Some(ActionKind::Fallible(ref action)) => action,
+                        _ => &empty_string
+                    }
+                };
+                if norm_util::check_funky_expression(action).is_in_curly_brackets() {
+                    return_err!(
+                        alternative.span,
+                        "the `<>` expression requires to explicitly assign fields' names to values");
```

Comment:
Nit: maybe something like this?

"using `<>` between curly braces (e.g., `{<>}`) only works when your parsed values have been given names (e.g., `<x:Foo>`, not just `<Foo>`)"